app/src/main/java/tech/ula/MainActivity.kt

```diff
@@ -593,6 +599,9 @@ class MainActivity : AppCompatActivity(), SessionListFragment.SessionSelection,
                 val step = getString(R.string.progress_copying_downloads)
                 updateProgressBar(step, "")
             }
+            is VerifyingAvailableStorage -> {
+                killProgressBar()
```

Comment:
I think that the progress bar should still be visible during this state. That way, in the successful case, the progress bar won't disappear between between the asset verification and filesystem extraction steps. While you're in there, can you reorder the `when` statement to reflect the actual order of steps? I.E verifying filesystem should be before extraction.